pkg/gpu/context.go

```
@@ -34,8 +34,11 @@ type systemContext struct {
 	// nvmlLib is the NVML library used to query GPU devices
 	nvmlLib nvml.Interface
 
-	// deviceSmVersions maps each device index to its SM (Compute architecture) version
-	deviceSmVersions map[int]int
+	// deviceSmVersions maps each device UUID to its SM (Compute architecture) version
+	deviceSmVersions map[string]uint32
+
+	// defaultSmVersion is the default SM version to use when the device is not found
+	defaultSmVersion uint32
```

is this really should be part of the context? why not making a global or even const? 

i think it is always best to keep the context as lean as possible as it tends to become a garbage for all kind of data (since the object is usually passed into multiple flows)

---

This should be part of the device manager after we do the refactor. However I don't think we can make it a const/global, as we want to have a value that is present on at least some device so that the kernel identification can be correct even on that fallback case (I think few hosts will have different model GPUs).

---

i see. actually i think the last assumption is wrong. even if our org will follow that pattern, I think generally the cloud envs for gpu are shifting towards multilple different devices in the same cluster

---

maybe in that case we shouldn't have the default at all? to avoid "masking" potential errors in the future